src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/VersionsController.kt

```diff
@@ -53,10 +53,11 @@ class VersionsController(private val session: Session,
 
     @GetMapping("version/{versionId}/snapshot/{snapshotId}")
     @ResponseBody
-    fun getSnapshotDetails(@PathVariable("versionId") versionId: Int,
-                    @PathVariable("snapshotId") snapshotId: String): ResponseEntity<String>
+    fun loadSnapshotDetails(@PathVariable("versionId") versionId: Int,
+                            @PathVariable("snapshotId") snapshotId: String): ResponseEntity<String>
     {
         val snapshotDetails = snapshotRepository.getSnapshotDetails(snapshotId, versionId, userId())
+        session.setSnapshotId(snapshotId)
```

Comment:
Changed this from 'get' to 'load' as it needs to set snapshot id for the session, don't just imply that this is a fetch of data with no side effects.